feat(Page): Add global timeout setting

radu125 opened the pull request:
References #1514

radu125 committed:
feat(Page): Add global timeout setting

References #1514

radu125 commented:
I signed the CLA.

aslushnikov commented:
Can you please eliminate the default 30000 timeout from the NavigatorWatcher?

https://github.com/GoogleChrome/puppeteer/blob/8e9c54a789c9d51f9861053b21565ad21a5ffe6f/lib/NavigatorWatcher.js#L41-L48

I'd do this by passing timeout explicitly to the NavigatorWatcher as one of the arguments in its constructor.

radu125 committed:
feat: pass timeout as separate argument

Remove the default 30000 timeout from NavigatorWatcher and pass the
timeout to the NavigatorWatcher constructor as a separate argument

radu125 commented:
I did it, but the AppVeyor build failed, and I can't figure out why that test failed.

aslushnikov commented on test/test.js:
```diff
@@ -464,6 +464,14 @@ describe('Page', function() {
   });
 
+  describe('Page.setDefaultNavigationTimeout', function() {
+    it('should work', async({page, server}) => {
```
We generally try hard not to reach out to private fields.
Let's drop this test – i think the other one you add is sufficient.

aslushnikov reviewed (commented):
@raduaron26 no worries about appveyour, some of our tests are flaky. 

Thank you for working on this. I have a minor comment, other than that it's good to go.

radu125 committed:
feat: remove unit test

Remove unit test in which the _defaultNavigationTimeout private field is
accessed

radu125 commented:
I removed the test. I'm glad I could contribute and thank you too for your help :)

aslushnikov commented on docs/api.md:
```diff
@@ -78,6 +78,7 @@
   * [page.select(selector, ...values)](#pageselectselector-values)
   * [page.setContent(html)](#pagesetcontenthtml)
   * [page.setCookie(...cookies)](#pagesetcookiecookies)
+  * [page.setDefaultNavigationTimeout(timeout)](#pagesetdefaultnavigationtimeouttimeout)
```
So this setting affects 5 methods: `page.goto`, `page.goBack`, `page.goForward`, `page.reload` and `page.waitForNavigation`.
Can you please:
- refer to the methods from the doc of the method, explaining that this method will change defaults for them, and that the default value is 30 seconds
- update documentation for the methods themselves, citing `page.setDefaultNavigationTimeout`.

This should make it more understandable for our clients.

aslushnikov reviewed (changes requested):
@raduaron26 I've just figured there's one last nit for the documentation before this should be merged. Thanks again for taking your time for this

radu125 committed:
feat: Update documentation

Update documentation regarding the page.setDefaultNavigationTimeout
functionality

radu125 commented:
I updated the documentation, let me know if you want me to make any tweaks to what I wrote.

aslushnikov reviewed (approved):
Thank you, this looks good.

stackflows commented:
@aslushnikov: This only affects navigation. waitForSelector() etc. should have a similar global "default" setting. Shall I write a feature request for this?

Peta5000 commented:
@stackflows Did you write a feature request for this? This feature would be so precious.

spinningarrow commented:
@Peta5000 it looks like #1514 is the issue for this but it is closed. I've added a comment there voting to reopen.

UPDATE: This is a part of #2079
